config/confighttp/confighttp.go

```diff
@@ -286,6 +288,14 @@ type ServerConfig struct {
 	ResponseHeaders map[string]configopaque.String `mapstructure:"response_headers"`
 }
 
+// NewDefaultServerConfig creates new ServerConfig with default values set
+func NewDefaultServerConfig() ServerConfig {
+
+	return ServerConfig{
+		ResponseHeaders: make(map[string]configopaque.String),
```

Comment:
This should use configtls.NewDefaultServerConfig and configauth.NewDefaultAuthentication and NewDefaultCORSConfig